iceoryx2-cli/iox2-service/src/commands.rs

```diff
@@ -66,3 +71,54 @@ pub fn details(service_name: String, filter: OutputFilter, format: Format) -> Re
     }
     Ok(())
 }
+
+pub fn discovery(
+    rate: u64,
+    publish_events: bool,
+    max_subscribers: usize,
+    send_notifications: bool,
+    max_listeners: usize,
+    format: Format,
+) -> Result<()> {
+    let discovery_config = DiscoveryConfig {
+        publish_events,
+        max_subscribers,
+        send_notifications,
+        max_listeners,
+        sync_on_initialization: true,
+        include_internal: false,
+    };
+
+    let mut service =
+        DiscoveryService::<ipc::Service>::create(&discovery_config, Config::global_config())
+            .map_err(|e| anyhow::anyhow!("failed to create service discovery service: {:?}", e))?;
+
+    println!("=== Service Started (rate: {}ms) ===", rate);
+    println!("{}", format.as_string(&discovery_config)?);
+
+    while !SignalHandler::termination_requested() {
```

Comment:
I think the iceoryx2 idiomatic approach would be to use a `WaitSet` here, attach the rate with `WaitSet::attach_interval` and then call `WaitSet::wait_and_process`.

The `WaitSet` already explicitly takes care of signal handling. It can be disabled in the waitset builder with `signal_handling_mode(SignalHandlingMode::Disabled)` but by default it is set to `SignalHandlingMode::HandleTerminationRequests`.